internal/server/remove_team_member_test.go

```diff
@@ -0,0 +1,275 @@
+	client.team.err = expectedError
+	template := &mockTemplate{}
+
+	w := httptest.NewRecorder()
+	r, _ := http.NewRequest("POST", "/teams/remove-member/123", strings.NewReader("selected[]=12&selected[]=45"))
+	r.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	r.AddCookie(&http.Cookie{Name: "test", Value: "val"})
+
+	err := removeTeamMember(client, template, "http://sirius")(w, r)
+	assert.Equal(expectedError, err)
+
+	assert.Equal(1, client.team.count)
+	assert.Equal(0, client.editTeam.count)
+}
+
+func TestPostRemoveTeamMemberBadData(t *testing.T) {
+	for name, data := range map[string]string{
+		"invalid":     "%1",
+		"non-numeric": "selected[]=string",
+	} {
+		t.Run(name, func(t *testing.T) {
+			assert := assert.New(t)
+
+			client := &mockRemoveTeamMemberClient{}
+			template := &mockTemplate{}
+
+			w := httptest.NewRecorder()
+			r, _ := http.NewRequest("POST", "/teams/remove-member/123", strings.NewReader(data))
+			r.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+			r.AddCookie(&http.Cookie{Name: "test", Value: "val"})
+
+			err := removeTeamMember(client, template, "http://sirius")(w, r)
+			assert.Equal(StatusError(http.StatusBadRequest), err)
+
+			assert.Equal(1, client.team.count)
+			assert.Equal(0, client.editTeam.count)
+		})
+	}
+}
+
+func TestPostRemoveTeamMemberIgnoresBadIds(t *testing.T) {
+	assert := assert.New(t)
+
+	client := &mockRemoveTeamMemberClient{}
+	client.team.data = generateTeamWithIds(12, 16, 45)
+	template := &mockTemplate{}
+
+	w := httptest.NewRecorder()
+	r, _ := http.NewRequest("POST", "/teams/remove-member/123", strings.NewReader("selected[]=19&selected[]=45"))
+	r.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	r.AddCookie(&http.Cookie{Name: "test", Value: "val"})
+
+	err := removeTeamMember(client, template, "http://sirius")(w, r)
+	assert.Nil(err)
+
+	assert.Equal(removeTeamMemberVars{
+		Path:      "/teams/remove-member/123",
+		SiriusURL: "http://sirius",
+		Team:      client.team.data,
+		Selected: []sirius.TeamMember{
+			{ID: 45},
+		},
+	}, template.lastVars)
+}
+
+func TestConfirmPostRemoveTeamMember(t *testing.T) {
+	assert := assert.New(t)
+
+	client := &mockRemoveTeamMemberClient{}
+	client.team.data = generateTeamWithIds(12, 16, 45)
+	template := &mockTemplate{}
+
+	w := httptest.NewRecorder()
+	r, _ := http.NewRequest("POST", "/teams/remove-member/123", strings.NewReader("selected[]=12&selected[]=45&confirm=true"))
+	r.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	r.AddCookie(&http.Cookie{Name: "test", Value: "val"})
+
+	err := removeTeamMember(client, template, "http://sirius")(w, r)
+	assert.Equal(RedirectError("/teams/123"), err)
+
+	assert.Equal(1, client.team.count)
+	assert.Equal(1, client.editTeam.count)
+
+	assert.Equal([]sirius.TeamMember{
+		{
+			ID: 16,
+		},
+	}, client.editTeam.lastTeam.Members)
+}
+
+func TestConfirmPostRemoveTeamMemberValidationError(t *testing.T) {
```

Comment:
this is actually returning a ClientError

Reply:
Fixed 👍